### ISSUE TITLE
Sa11y bugs + fixes

### ISSUE THREAD

**Issue by adamchaboryk** [joomla-projects/joomla-a11y-checker#53]
Hi Joomla team,

Just passing along some bugs and respective fixes. If below isn't clear enough, you can reference my development branch. Secondly, thank you all so much for working on the rest of the vanilla JS port - I brought back your code to my repo and i'm officially free of jQuery too! I'll be sure to pass along any future bug fixes and enhancements.

## CSS fixes
- Replace all CSS class instances of `“sa11y-error-heading”` with `“sa11y-error-border”`.  Otherwise the font colour conflicts with dark themes and the red error text will be flagged for contrast issues. Oops! 
     - For fake heading detection ruleset, replace `“sa11y-error-heading”` with “sa11y-warning-border”.
     - For semantic headings within `<table>` ruleset, remove the following line: ` $el.parentNode.classList.add("sa11y-error-heading");`, otherwise too many red borders.
     - Once you've replaced all instances, you can remove "sa11y-error-heading" from CSS file, and clearEverything/resetAll function.
- The CSS reset for letter-spacing doesn’t seem to work if left as unset. Change letter-spacing: unset !important, to letter-spacing: normal !important;

### Fake headings
For fake heading detection ruleset, within both if statements: add `!$el.closest("table")` so it doesn’t give conflicting advice to add heading tags within semantic tables. 
` if (!$el.closest("table") && boldtext.length <= 120) {...` and `if (!$el.closest("table") && $el.textContent.length <= 120 && tagName.charAt(0) !== "H") {`

### Readability
Readability helper affects fake heading detection.  Not the most elegant fix, but only add invisible period utility when paragraph length is greater than 120 characters (arbitrary value for fake heading ruleset). Otherwise, shorter fake headings will get extra HTML markup that conflicts with fake heading logic.

````javascript 
this.$readability.forEach($el => {
                var listText = $el.textContent;
                if (listText.length >= 120) {
                    if (listText.charAt(listText.length - 1) !== ".") {
                        $el.insertAdjacentHTML("beforeend", "<span class='sa11y-readability-period sa11y-visually-hidden'>.</span>");
                    }
                }
            });
 ````

### Link check (advanced)
Need a surrounding if (linkText !== null), otherwise empty links get both a warning and error flag.

````javascript
if (linkText !== null) {
                    if (seen[linkTextTrimmed] && linkTextTrimmed.length !== 0) {
                        if (seen[href]) {
                            //Nothing
                        } else {
                            this.warningCount++;
                            el.classList.add("sa11y-warning-text");
                            el.insertAdjacentHTML(
                                'afterend',
                                Sa11yAnnotate(sa11yWarning, M["linkIdenticalName"](linkText), true));
                        }
                    } else {
                        seen[linkTextTrimmed] = true;
                        seen[href] = true;
                    }
}
````
### Alt text check
Any text within `<noscript>` is picked up by `innerText` and provides wrong tooltip message. E.g.`<a href="#"><img src="#" alt="Homepage"><noscript>asdfg</noscript></a>` 

To fix, I used your `fnIgnore` function on any if statement that has anchor and textContent.  E.g.
`if (fnIgnore($el.closest('a[href]'), "noscript").textContent.trim().length >= 1) {`

**Comment by brianteeman** [comment]
Thanks @adamchaboryk working through the changes now

**Comment by brianteeman** [comment]
- [x] Css
- [x] Fake Headings
- [x] Readability
- [x] Link Check Advanced - see https://github.com/ryersondmp/sa11y/issues/25
- [x] Alt text

**Comment by brianteeman** [comment]
closed - all now in #53 thank you
